api/src/extensions/lib/sync-extensions.ts

```diff
@@ -0,0 +1,94 @@
+import { NESTED_EXTENSION_TYPES } from '@directus/extensions';
+import { ensureExtensionDirs } from '@directus/extensions/node';
+import mid from 'node-machine-id';
+import { createWriteStream } from 'node:fs';
+import { mkdir } from 'node:fs/promises';
+import { uptime } from 'node:os';
+import { dirname, join, normalize, resolve, sep } from 'node:path';
+import { pipeline } from 'node:stream/promises';
+import Queue from 'p-queue';
+import env from '../../env.js';
+import logger from '../../logger.js';
+import { getMessenger } from '../../messenger.js';
+import { getStorage } from '../../storage/index.js';
+import { getExtensionsPath } from './get-extensions-path.js';
+import { SyncStatus, getSyncStatus, setSyncStatus } from './sync-status.js';
+import { getSyncTimestamp, setSyncTimestamp } from './sync-timestamp.js';
+
+export const syncExtensions = async () => {
+	const messenger = getMessenger();
+
+	const isPrimaryProcess =
+		String(process.env['NODE_APP_INSTANCE']) === '0' || process.env['NODE_APP_INSTANCE'] === undefined;
```

Comment:
Is this really required? Values in `process.env` should always be a string 🤔 

```suggestion
	process.env['NODE_APP_INSTANCE'] === '0' || process.env['NODE_APP_INSTANCE'] === undefined;
```

Reply:
Technically not, but our `env.ts` handling does some magic in that global space so better safe than sorry here